Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1426,41 +1426,75 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CFNotificationCenter
 }
 #endif
 
-static void addUserInstalledFontURLs(NSString *path, Vector<URL>& fontURLs)
+static Vector<SandboxExtension::Handle> sandboxExtensionsForUserInstalledFonts(const Vector<URL>& fontPathURLs, std::optional<audit_token_t> auditToken)
+{
+    Vector<SandboxExtension::Handle> handles;
+    for (auto& fontPathURL : fontPathURLs) {
+        std::optional<SandboxExtension::Handle> sandboxExtensionHandle;
+        if (auditToken)
+            sandboxExtensionHandle = SandboxExtension::createHandleForReadByAuditToken(fontPathURL.fileSystemPath(), *auditToken);
+        else
+            sandboxExtensionHandle = SandboxExtension::createHandle(fontPathURL.fileSystemPath(), SandboxExtension::Type::ReadOnly);
+        if (sandboxExtensionHandle)
+            handles.append(WTFMove(*sandboxExtensionHandle));
+    }
+    return handles;
+}
+
+static void addUserInstalledFontURLs(NSString *path, HashMap<String, URL>& fontURLs, Vector<URL>& sandboxExtensionURLs)
 {
+    RetainPtr pathURL = [NSURL fileURLWithPath:path];
```

Comment:
Let's use `adoptNS()` and `initFileURLWithPath` instead. We try to avoid autoreleasing whenever possible.